infra/scripts/replace-running-components.sh

```diff
@@ -0,0 +1,96 @@
+function on-vch() {
+    sshpass -ppassword ssh -oUserKnownHostsFile=/dev/null -oStrictHostKeyChecking=no root@$VCH_IP -C $@ 2>/dev/null
+}
+
+# SCPs the component in $1 to the VCH, plops it in place, and brutally kills the previous running process
+function replace-component() {
+    sshpass -p 'password' scp -oUserKnownHostsFile=/dev/null -oStrictHostKeyChecking=no $VIC_DIR/bin/$1 root@$VCH_IP:/tmp/$1
+    on-vch mv /sbin/$1 /tmp/old-$1
+    on-vch mv /tmp/$1 /sbin/$1
+    on-vch chmod 755 /sbin/$1
+    pid=$(on-vch ps -e --format='pid,args' | grep $1 | grep -v grep | awk '{print $1}')
+    on-vch kill -9 $pid
+    on-vch rm -f /tmp/old-$1
+}
+
+function vch-name-is-ambiguous () {
+    [ $($VIC_DIR/bin/vic-machine-linux ls --target=$target --user=$username --password=$password --thumbprint=$(get-thumbprint)  | grep $VIC_NAME | wc -l) -ne 1 ] && return 0 || return 1
+}
+
+# Check GOVC vars
+if [[ ! $(govc ls 2>/dev/null) ]]; then
+    echo "ERROR:"
+    echo "GOVC environment variables are required to use this command. Set the necessary variables to allow govc to connect to your vSphere deployment:";
+    echo "GOVC_USERNAME: username on vSphere target"
+    echo "GOVC_PASSWORD: password on vSphere target"
+    echo "GOVC_URL: IP or FQDN of your vSphere target"
+    echo "GOVC_INSECURE: set to 1 to disable tls verify when using govc to talk to vSphere"
+    exit 1
+fi
+
+# Check for our one required argument
+if [[ ! -v VIC_NAME ]] && [[ ! -v VIC_ID ]]; then
+    echo "Please set one of the following environment variables to specify the VCH which you would like to reconfigure:"
+    echo "VIC_NAME: name of VCH; matches --name argument for vic-machine"
+    echo "VIC_ID: ID of VCH, as displayed in output of vic-machine ls"
+    exit 1
+fi
+
+username=$(govc env | grep GOVC_USERNAME | cut -d= -f2)
+password=$(govc env | grep GOVC_PASSWORD | cut -d= -f2)
+target=$(govc env | grep GOVC_URL | cut -d= -f2)
+
+
+if [[ ! -v VIC_ID ]] && [[ $(vch-name-is-ambiguous) ]]; then
+    echo "The provided VIC name is ambiguous; please choose the correct VCH ID from the output below and assign it to the environment variable VIC_ID, e.g., export VIC_ID=12"
+    $VIC_DIR/bin/vic-machine-linux ls --target $GOVC_URL --user $GOVC_USER --password=$GOVC_PASSWORD --thumbprint=$(get-thumbprint)
+    exit 1
+fi
+
+if [[ ! -v VIC_ID ]]; then
+    VIC_ID=$($VIC_DIR/bin/vic-machine-linux ls --target=$target --user=$username --password=$password --thumbprint=$(get-thumbprint) | grep $VIC_NAME | awk '{print $1}')
+fi
+
+$VIC_DIR/bin/vic-machine-linux debug --target=$target --id=$VIC_ID --user=$username --password=$password --thumbprint=$(get-thumbprint)
+
+VCH_IP="$(govc vm.ip $VIC_NAME)"
+
+
+if [[ $1 == "" ]]; then # replace everything
+    for x in port-layer-server docker-engine-server vicadmin vic-init; do
+        replace-component $x
+    done
+else
+    for x in $@; do # replace provided list
+        replace-component $x
+    done
+fi
+
+on-vch vic-init &
```

Comment:
This really shouldn't be necessary unless configured at a debug level that causes vic-init to exit instead of relaunching components (think that's debug>=3)

I'm not sure what will happen here, but there is a risk of multiple vic-init's running - I don't know that there's anything that prevents that.

I'd suggest using `systemctl restart vic-init.service` to perform a restart if necessary.
I'd also ask you to follow up on why the restart is needed.